nomt/src/commit/worker.rs

```diff
@@ -362,19 +397,80 @@ impl<H: NodeHasher> RangeCommitter<H> {
         }
     }
 
-    fn conclude(mut self, output: &mut WorkerOutput) -> Option<WritePass<ShardIndex>> {
-        // PANIC: walker was configured with a parent page.
-        let (new_nodes, diffs) = match self.page_walker.conclude(&mut self.write_pass) {
-            Output::Root(_, _) => unreachable!(),
-            Output::ChildPageRoots(new_nodes, diffs) => (new_nodes, diffs),
-        };
+    fn reattempt_advance(&mut self, seeker: &mut NewSeeker, output: &mut WorkerOutput) {
+        // UNWRAP: guaranteed by behavior of seeker / commit / advance.
+        let SavedAdvance {
+            ops,
+            seek_result,
+            batch_size,
+        } = self.saved_advance.take().unwrap();
+        self.attempt_advance(seeker, output, seek_result, ops, batch_size);
+    }
+
+    fn commit(
+        mut self,
+        seeker: &mut NewSeeker,
+        output: &mut WorkerOutput,
+    ) -> anyhow::Result<Option<WritePass<ShardIndex>>> {
+        let mut start_index = self.range_start;
+        let mut pushes = 0;
+        let mut skips = 0;
+
+        let mut conclude_needs_extra = false;
```

Comment:
I am not sure if I fully understand the purpose of this variable. In what case does it become true?

Reply:
When we call `page_walker.conclude()` and an extra page becomes needed. Though I refactored this somewhat upstack and it should become more legible.